api/v1alpha3/awscluster_types.go

```diff
@@ -82,6 +86,10 @@ type AWSClusterSpec struct {
 	// Bastion contains options to configure the bastion host.
 	// +optional
 	Bastion Bastion `json:"bastion"`
+
+	// PrincipalRef is a reference to a principal to be used when reconciling this cluster
+	// +optional
+	PrincipalRef *corev1.ObjectReference `json:"principalRef,omitempty"`
```

Comment:
Don't use ObjectReference here. We do not want the `Namespace`, `FieldPath`, `ResourceVersion`, `APIVersion` and `UID` fields.

Create a `PrincipalReference` struct with only `Kind` and `Name` fields.

Reply:
just to give more context https://github.com/kubernetes/kubernetes/https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/vmware/govmomi/find/finder.go#L5349-L5365

Reply:
Makes sense, but I think we should keep `Namespace` for `AWSServiceAccountPrincipal` implementation

Reply:
`AWSServiceAccountPrincipal` is intended to be namespace scoped as well, so shouldn't require `Namespace` on principalRef too. Given that we're not implementing yet, we can come back to it. We can always add fields to the new reference type, but removing them afterwards will be more problematic.